core/templates/dev/head/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -16,122 +16,140 @@
  * @fileoverview Factory for creating new frontend instances of Interaction
  * domain objects.
  */
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { AnswerGroupObjectFactory } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { HintObjectFactory } from 'domain/exploration/HintObjectFactory';
+import { OutcomeObjectFactory } from 'domain/exploration/OutcomeObjectFactory';
+import { SolutionObjectFactory } from
+  'domain/exploration/SolutionObjectFactory';
+
+export class Interaction {
+  answerGroups;
```

Comment:
Shouldn't these by preceded by underscores?

Reply:
Just curious to know why do we need underscores?

Reply:
These are private variables. In other object factories, private variables are preceded by underscores.

Reply:
I think we cannot do that here. Please take a look at the test file -- I'm pasting some section from the same.
```
testInteraction = iof.createFromBackendDict(interactionDict);
  });

  it('should correctly set the new ID', () => {
    expect(testInteraction.id).toEqual('interaction_id');
    testInteraction.setId('new_interaction_id');
    expect(testInteraction.id).toEqual('new_interaction_id');
  });
```
testinteraction is the object of `Interaction` and we are comparing `testInteraction.id` which implies id is not a private property and we should not use `_id`. Incase we use, we have to apply underscore everywhere wherever these properties are used.

Reply:
Oh, in all other object factories though, they are private. Then, I guess for this, no need to change it.